script.js

```diff
@@ -1,4 +1,294 @@
+/* Beat That - Base
+Requirements
+1. There are 2 players and players take turns.
+2. When a player clicks Submit, the game rolls 2 dice and shows the dice rolls, for example 3 and 6.
+3. The player picks the order of the dice they want. For example, if they wanted the number 63, they would specify that the 2nd dice goes first. You can choose how the player        
+   specifies dice order.
+4. After both players have rolled and chosen dice order, the player with the higher combined number wins.
+
+Objectives:
+1. 1v1 player vs player dice roll game &&
+2. Roll 2 dice and concentate the 2 dice number together &&
+3. Players can choose order of dice to concentate (dice 1 + dice 2 || dice 2 + dice 1)
+
+What is required?
+1. Dice Roll helper function
+2. Global states for different gamemode
+3. Game function
+4. Instructions for user
+*/
+
+//GLOBAL VARIABLES
+var currentGameMode = "welcomeToBeatThat"; //Starting Game Mode
+var player1 = ""; //Player 1 IGN
+var player2 = ""; //Player 2 IGN
+
+var userChosenDiceOrder = ""; //User desired dice order, either 1 or 2 (input = userDiceOrder)
+
+// player 1 dice results, store in array for later use, prevent another diceroll to reset results too.
+var player1dice1Arr = [];
+var player1dice2Arr = [];
+// player 2 dice results, store in array for later use, prevent another diceroll to reset results too.
+var player2dice1Arr = [];
+var player2dice2Arr = [];
+
+var player1Score = ""; //Player 1 score at the end of turn
+var player2Score = ""; //Player 2 score at the end of turn
```

Comment:
nice and clear global variables, well commented. i would say that if the variable name is descriptive enough, such as the case of `player1Score`, the comment is unnecessary. this is another point for why descriptive variable names are desired as your code becomes self-documenting

i don't see the point of having a `player1dice1Arr` and `player1dice2Arr`, similarly for player 2. seems like in the rest of your code, these arrays only store a max of 1 value, which defeats the purpose of an array. maybe it was an intentional design choice but from the rest of your code, it seems that it would be better if we have two arrays, `player1Rolls` and `player2Rolls`, and we push the dice rolls into the relevant array for each player